backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -1390,6 +1407,60 @@ public ExperimentRunPaginationDTO findExperimentRuns(FindExperimentRuns queryPar
     return hyperparametersMap;
   }
 
+  private Map<String, Map<String, CodeBlob>> getExperimentRunCodeBlobMap(
+      Session session, List<String> expRunIds) throws InvalidProtocolBufferException {
+    String queryBuilder =
+        "SELECT vme.experimentRunEntity.id, vme.versioning_location, gcb, ncb, pdcb "
+            + " From VersioningModeldbEntityMapping vme LEFT JOIN GitCodeBlobEntity gcb ON vme.blob_hash = gcb.blob_hash "
+            + " LEFT JOIN NotebookCodeBlobEntity ncb ON vme.blob_hash = ncb.blob_hash "
+            + " LEFT JOIN PathDatasetComponentBlobEntity pdcb ON ncb.path_dataset_blob_hash = pdcb.id.path_dataset_blob_id "
+            + " WHERE vme.versioning_blob_type = :versioningBlobType AND vme.experimentRunEntity.id IN (:expRunIds)";
+    Query query = session.createQuery(queryBuilder);
+    query.setParameter("versioningBlobType", Blob.ContentCase.CODE.getNumber());
+    query.setParameterList("expRunIds", expRunIds);
+    LOGGER.debug("Final experimentRuns code config blob final query : {}", query.getQueryString());
+    List<Object[]> codeBlobEntities = query.list();
+    LOGGER.debug("Final experimentRuns code config list size : {}", codeBlobEntities.size());
+    Map<String, Map<String, CodeBlob>> expRunCodeBlobMap = new LinkedHashMap<>();
```

Comment:
same as above comment on the contents of the map

Reply:
Comment added